### DISCUSSION TITLE
Display Ecto Errors for Live Resources

### DISCUSSION THREAD

**Discussion by sneha-saj** [naymspace/backpex#477] [Q&A] [answered]
How can I display errors returned by Ecto changesets in my app? For example, when a unique constraint error occurs (e.g., attempting to create an entry with a duplicate name), it should trigger an error message. Currently, my application gets stuck on the creation screen without displaying any error.

**Comment by Flo0807** [comment, marked as answer]
You have to add the error under the name of a field you are displaying.

For example, if you have a `username` field like this

```elixir
username: %{
  module: Backpex.Fields.Text,
  label: "Username"
},
```

you need to add the error under the `username` key to the changeset.

[`Ecto.Changeset.unique_constraint/3`](https://hexdocs.pm/ecto/Ecto.Changeset.html#unique_constraint/3) also has an `error_key` option, which in this case should be `:username`.

@sneha-saj
